sklearn/linear_model/stochastic_gradient.py

```diff
@@ -1341,3 +1354,416 @@ def __init__(self, loss="squared_loss", penalty="l2", alpha=0.0001,
+        the data.  If int, random_state is the seed used by the random number
+        generator; If RandomState instance, random_state is the random number
+        generator; If None, the random number generator is the RandomState
+        instance used by `np.random`.
+
+    learning_rate : string, optional
+        The learning rate schedule:
+
+        - 'constant': eta = eta0
+        - 'optimal': eta = 1.0 / (alpha * (t + t0)) [default]
+        - 'invscaling': eta = eta0 / pow(t, power_t)
+
+        where t0 is chosen by a heuristic proposed by Leon Bottou.
+
+    eta0 : double
+        The initial learning rate for the 'constant' or 'invscaling'
+        schedules. The default value is 0.0 as eta0 is not used by the
+        default schedule 'optimal'.
+
+    power_t : double
+        The exponent for inverse scaling learning rate [default 0.5].
+
+    warm_start : bool, optional
+        When set to True, reuse the solution of the previous call to fit as
+        initialization, otherwise, just erase the previous solution.
+
+    average : bool or int, optional
+        When set to True, computes the averaged SGD weights and stores the
+        result in the ``coef_`` attribute. If set to an int greater than 1,
+        averaging will begin once the total number of samples seen reaches
+        average. So ``average=10`` will begin averaging after seeing 10
+        samples.
+
+    Attributes
+    ----------
+    coef_ : array, shape (1, n_features)
+        Weights assigned to the features.
+
+    offset_ : array, shape (1,)
+        Offset used to define the decision function from the raw scores.
+        We have the relation: decision_function = score_samples - offset_.
+        The offset is the opposite of intercept_ and is provided for
+        consistency with other outlier detection algorithms.
+
+    n_iter_ : int
+        The actual number of iterations to reach the stopping criterion.
+
+    loss_function_ : concrete ``LossFunction``
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn import linear_model
+    >>> X = np.array([[-1, -1], [-2, -1], [1, 1], [2, 1]])
+    >>> clf = linear_model.SGDOneClassSVM()
+    >>> clf.fit(X)
+    ... #doctest: +NORMALIZE_WHITESPACE
+    SGDOneClassSVM(average=False, eta0=0.0, fit_intercept=True,
+            learning_rate='optimal', max_iter=None, nu=0.5,
+            power_t=0.5, random_state=None, shuffle=True,
+            tol=None, verbose=0, warm_start=False)
+
+    >>> print(clf.predict([[4, 4]]))
+    [-1]
+
+    See also
+    --------
+    OneClassSVM
+
+    """
+
+    loss_functions = {"hinge": (Hinge, 1.0)}
+
+    def __init__(self, nu=0.5, fit_intercept=True, max_iter=None, tol=None,
+                 shuffle=True, verbose=0, random_state=None,
+                 learning_rate="optimal", eta0=0.0, power_t=0.5,
+                 warm_start=False, average=False):
+
+        if not(0 < nu <= 1):
+            raise ValueError("nu must be in (0, 1]")
```

Comment:
We usually avoid checking  parameters in the `__init__`, since if the value is changed with `set_params`, it has to be checked again during the fit.

Reply:
I agree but there is a `self.validate_params` in `BaseSGD.__init__`. If I remove the check about `nu` in `SGDOneClassSVM.__init__` no error message or a non-informative error message is raised concerning the value of `alpha` which is not a parameter of the `SGDOneClassSVM`